boilingcore/boilingcore.go

```diff
@@ -47,6 +47,10 @@ type State struct {
 
 // New creates a new state based off of the config
 func New(config *Config) (*State, error) {
+	if config.FileExtension == "" {
```

Comment:
Note: this preserves backwards compatibility for anyone who is importing this package as a library instead of using the CLI directly.